integration/query_fuzz_test.go

```diff
@@ -634,6 +634,36 @@ func TestProtobufCodecFuzz(t *testing.T) {
 	runQueryFuzzTestCases(t, ps, c1, c2, now, start, end, scrapeInterval, 1000)
 }
 
+var sampleNumComparer = cmp.Comparer(func(x, y model.Value) bool {
+	if x.Type() != y.Type() {
+		return false
+	}
+
+	vx, xvec := x.(model.Vector)
+	vy, yvec := y.(model.Vector)
+
+	if xvec && yvec {
+		return len(vx) == len(vy)
+	}
+
+	mx, xmat := x.(model.Matrix)
+	my, ymat := y.(model.Matrix)
+
+	mxSamples := 0
+	mySamples := 0
+
+	if xmat && ymat {
+		for i := 0; i < len(mx); i++ {
+			mxSamples += len(mx[i].Values)
+		}
+		for i := 0; i < len(my); i++ {
+			mySamples += len(my[i].Values)
+		}
+	}
+
+	return mxSamples == mySamples
```

Comment:
Maybe we can change this a bit to basically still compare all the results and just ignore the series - as we know if the results are the same, the series returned can be random?

```suggestion
	if xvec && yvec && len(vx) == len(vy) {
		for i, sample := range vx {
			vy[i].Metric = nil
			sample.Metric = nil
			if !vy[i].Equal(sample) {
				return false
			}
		}
		return true
	}

	mx, xmat := x.(model.Matrix)
	my, ymat := y.(model.Matrix)

	if xmat && ymat && len(mx) == len(my) {
		for i, ss := range mx {
			my[i].Metric = nil
			ss.Metric = nil
			if len(my[i].Histograms) != len(ss.Histograms) {
				return false
			}

			for k, histogram := range ss.Histograms {
				if !my[i].Histograms[k].Equal(&histogram) {
					return false
				}
			}

			if len(my[i].Values) != len(ss.Values) {
				return false
			}

			for k, v := range ss.Values {
				if !my[i].Values[k].Equal(&v) {
					return false
				}
			}
		}

		return true
	}

	return false
```